Library1/Library.fs

```diff
@@ -22,9 +22,389 @@ module FetchData =
+
+    interface ILayoutComponent with
+        member this.Body with get() = this.Body
+                          and set(value) = this.Body <- value
+
+type SurveyPrompt () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+        builder.OpenElement(0, "div")
+        builder.AddAttribute(1, "class", "alert alert-survey")
+        builder.AddAttribute(2, "role", "alert")
+        builder.AddContent(3, "\n    ")
+        builder.OpenElement(4, "span")
+        builder.AddAttribute(5, "class", "glyphicon glyphicon-ok-circle")
+        builder.AddAttribute(6, "aria-hidden", "true")
+        builder.CloseElement()
+        builder.AddContent(7, "\n    ")
+        builder.OpenElement(8, "strong")
+        builder.AddContent(9, this.Title)
+        builder.CloseElement()
+        builder.AddContent(10, "\n\n    Please take our\n    ")
+        builder.OpenElement(11, "a")
+        builder.AddAttribute(12, "target", "_blank")
+        builder.AddAttribute(13, "class", "alert-link")
+        builder.AddAttribute(14, "href", "https://go.microsoft.com/fwlink/?linkid=870381")
+        builder.AddContent(15, "\n        brief survey\n    ")
+        builder.CloseElement()
+        builder.AddContent(16, "\n    and tell us what you think.\n")
+        builder.CloseElement()
+        builder.AddContent(17, "\n\n")
+
+    // This is to demonstrate how a parent component can supply parameters
+    member val Title : string = Unchecked.defaultof<string> with get, set
+
+namespace BlazorApp1.Pages
+
+open System
+open System.Collections.Generic
+open System.Linq
+open System.Threading.Tasks
+open System.Net.Http
+open Microsoft.AspNetCore.Blazor
+open Microsoft.AspNetCore.Blazor.Components
+open Microsoft.AspNetCore.Blazor.Layouts
+open Microsoft.AspNetCore.Blazor.Routing
+open BlazorApp1
+open BlazorApp1.Shared
+
+// NOTE: _ViewImports.cshtml appear to be required in the BlazorApp.
+[<Microsoft.AspNetCore.Blazor.Layouts.LayoutAttribute(typeof<MainLayout>)>]
+type _ViewImports () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+
+[<Microsoft.AspNetCore.Blazor.Layouts.LayoutAttribute(typeof<MainLayout>)>]
+[<Microsoft.AspNetCore.Blazor.Components.RouteAttribute("/")>]
+type Index () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+        builder.AddContent(0, "\n")
+        builder.OpenElement(1, "h1")
+        builder.AddContent(2, "Hello, world!")
+        builder.CloseElement()
+        builder.AddContent(3, "\n\nWelcome to your new app.\n\n")
+        builder.OpenComponent<SurveyPrompt>(4)
+        builder.AddAttribute(5, "title", "How is Blazor working for you?")
+        builder.CloseComponent()

```

Comment:
The generated code used:

``` fsharp
builder.OpenElement(4, "surveyprompt")
builder.AddAttribute(5, "title", "How is Blazor working for you?")
builder.CloseElement()
```

That didn't work, but using the typed `OpenComponent` did.